### PR TITLE
[EventEngine] [reland] Migrate `chttp2_server` to use EE DNSResolver

### PR DESCRIPTION
**Pull request by yijiem**

### CODE REVIEW

File: src/core/ext/transport/chttp2/server/chttp2_server.cc [hunk 1]
```diff
@@ -966,6 +966,7 @@ grpc_error_handle Chttp2ServerAddPort(Server* server, const char* addr,
                                                     args_modifier);
   }
   *port_num = -1;
+  absl::StatusOr<std::vector<grpc_resolved_address>> resolved_or;
```
Comment (drfloob) [inline src/core/ext/transport/chttp2/server/chttp2_server.cc:969]: Please see https://abseil.io/tips/181 for tips on avoiding the `_or` suffix.

File: src/core/ext/transport/chttp2/server/chttp2_server.cc [hunk 2]
```diff
@@ -974,13 +975,21 @@ grpc_error_handle Chttp2ServerAddPort(Server* server, const char* addr,
...
   grpc_error_handle error = [&]() {
     grpc_error_handle error;
-    if (absl::ConsumePrefix(&parsed_addr_unprefixed, kUnixUriPrefix) ||
-        absl::ConsumePrefix(&parsed_addr_unprefixed, kUnixAbstractUriPrefix) ||
-        absl::ConsumePrefix(&parsed_addr_unprefixed, kVSockUriPrefix)) {
-      absl::StatusOr<EventEngine::ResolvedAddress> result =
-          grpc_event_engine::experimental::URIToResolvedAddress(parsed_addr);
-      GRPC_RETURN_IF_ERROR(result.status());
-      results->push_back(*result);
+    // TODO(ladynana, yijiem): this code does not handle address URIs correctly:
```
Comment (drfloob) [inline src/core/ext/transport/chttp2/server/chttp2_server.cc:978]: I would add here that part of this fix needs to be adding API documentation that `grpc_server_add_http2_port` expects a valid URI.

File: src/core/ext/transport/chttp2/server/chttp2_server.cc [hunk 3]
```diff
@@ -1001,6 +1010,12 @@ grpc_error_handle Chttp2ServerAddPort(Server* server, const char* addr,
         }
       }
     }
+    if (resolved_or.ok()) {
```
Comment (drfloob) [inline src/core/ext/transport/chttp2/server/chttp2_server.cc:1013]: If it's not OK, I think it would have returned already.

Comment (yijiem) [reply src/core/ext/transport/chttp2/server/chttp2_server.cc:1013]: The code path that populates `resolved` may not be ran. In that case `resolved` is default constructed and has no OK status.

File: src/core/ext/transport/chttp2/server/chttp2_server.cc (hunk 1 above)
Comment (yijiem) [reply src/core/ext/transport/chttp2/server/chttp2_server.cc]: done.

File: src/core/ext/transport/chttp2/server/chttp2_server.cc (hunk 2 above)
Comment (yijiem) [reply src/core/ext/transport/chttp2/server/chttp2_server.cc:978]: done.

File: src/core/ext/transport/chttp2/server/chttp2_server.cc (hunk 3 above)
Comment (drfloob) [reply src/core/ext/transport/chttp2/server/chttp2_server.cc]: I see it now, thanks.

**Review by drfloob — APPROVED** [review]

**Pull request closed without merging**